optuna/visualization/matplotlib/_pareto_front.py

```diff
@@ -74,13 +81,54 @@ def objective(trial):
     """
 
     _imports.check()
-
-    if len(study.directions) == 2:
-        return _get_pareto_front_2d(study, target_names, include_dominated_trials, axis_order)
-    elif len(study.directions) == 3:
-        return _get_pareto_front_3d(study, target_names, include_dominated_trials, axis_order)
+    if axis_order:
+        warnings.warn(
+            "``axis_order`` option is deprecated. This feature will be removed in the future.",
+            DeprecationWarning,
+        )
+    trials = _get_trials(study, include_dominated_trials)
+    if targets is not None and axis_order is not None:
+        raise ValueError(
+            "Using both ``targets`` and ``axis_order`` is not supported. "
+            " Use either `targets` or `axis_order`."
+        )
+    if targets is None:
+        if len(study.directions) == 2:
+            targets = _targets_default_2d
+        elif len(study.directions) == 3:
+            targets = _targets_default_3d
+        else:
+            raise ValueError(
+                "``plot_pareto_front`` function only supports 2 or 3 objective"
+                " studies when using ``targets`` is ``None``. Please use`targets`"
```

Comment:
How about clarifying why the `targets` argument is necessary?

```suggestion
                " studies when using ``targets`` is ``None``. Please specify the used values with `targets`."
```